### PR TITLE
Instrumentation 

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
This PR adds a variety of statistics and analytics for tracking the state of managers and the server itself. A few notables:
- The manager has had some internal variables renamed to hopefully be more understandable at first glance.
- The managers periodically record a log so that we can observe their over time performance, this happens every manager heartbeat (12 minutes by default) and on start/stop. Instantaneous parameters have also been added to the primary manager table.
- The server now records a variety of statistics every heartbeat (30 minutes by default), this includes database size, row counts, row information, etc.
- @doaa-altarawy I added a variety of custom queries. These queries are likely better off in advanced when we get to it, but is their current spot ok for now?

## Status
- [ ] Changelog updated
- [x] Ready to go

### CODE REVIEW

File: qcfractal/tests/test_server.py [hunk 1]
```diff
@@ -22,64 +22,18 @@ def test_server_information(test_server):
...
 
+    mol_count = server_info["counts"]["molecule"]
 
-def test_molecule_socket(test_server):
+    client.add_molecules([ptl.Molecule.from_data("Ne 0 0 3.1415")])
 
-    mol_api_addr = test_server.get_address("molecule")
-    water = ptl.data.get_molecule("water_dimer_minima.psimol")
+    test_server.update_server_log()
+    test_server.update_public_information()
```
Comment (mattwelborn) [inline qcfractal/tests/test_server.py]: Were these tests intentionally removed?

File: qcfractal/queue/executor_adapter.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def _submit_task(self, task_spec: Dict[str, Any]) -> Tuple[Hashable, Any]:
         task = self.client.submit(func, *task_spec["spec"]["args"], **task_spec["spec"]["kwargs"])
         return task_spec["id"], task
 
-    def count_running_workers(self) -> int:
+    def count_running_tasks(self) -> int:
         # This is always "running", even if there are no tasks since its running locally
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcfractal/queue/executor_adapter.py]: This was always broken actually, I went ahead and dug out a fix.

File: qcfractal/queue/managers.py [hunk 3]
```diff
@@ -25,17 +25,22 @@ class QueueStatistics(BaseModel):
...
+
     # Dynamic quantities
     total_successful_tasks: int = 0
     total_failed_tasks: int = 0
-    total_core_hours: float = 0
-    total_core_hours_consumed: float = 0
-    total_core_hours_possible: float = 0
+    total_worker_walltime: float = 0.0
+    total_task_walltime: float = 0.0
+    maximum_possible_walltime: float = 0.0
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcfractal/queue/managers.py]: Something else, marked it as experimental. An attempt to detail resource queue pressure.

File: qcfractal/tests/test_server.py (hunk 1 above)
Comment (dgasmith) [reply qcfractal/tests/test_server.py]: Yes, highly redundant from other areas.

File: qcfractal/queue/managers.py [hunk 4]
```diff
@@ -218,6 +229,7 @@ def __init__(self,
             # Tell the server we are up and running
             payload = self._payload_template()
             payload["data"]["operation"] = "startup"
+            # payload["data"]["configuration"] = self.configuration
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcfractal/queue/managers.py]: No, good catch.

File: qcfractal/server.py [hunk 5]
```diff
@@ -380,6 +384,15 @@ def start_manager():
             heartbeats.start()
             self.periodic["heartbeats"] = heartbeats
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcfractal/server.py]: This goes back to the `ByteSize` item in Elemental. Can we move this discussion there?

[the remark this answers is not in the document]
Comment (mattwelborn) [reply qcfractal/server.py]: Yep.

File: qcfractal/storage_sockets/db_queries.py [hunk 6]
```diff
@@ -1,24 +1,32 @@
...
-    def __init__(self, max_limit=1000):
+    def __init__(self, database_name, max_limit=1000):
+        self.database_name = database_name
         self.max_limit = max_limit
 
+    def __init_subclass__(cls, **kwargs):
+        if cls not in QUERY_CLASSES:
+            QUERY_CLASSES.add(cls)
+        super().__init_subclass__(**kwargs)
+
```
Comment (doaa-altarawy) [inline qcfractal/storage_sockets/db_queries.py]: Very nice :)

**Pull request merged**